span.go

```diff
@@ -58,3 +62,30 @@ type Span interface {
 	// TraceContext.
 	AddToGoContext(goCtx context.Context) (Span, context.Context)
 }
+
+// See Span.Log(). Every LogData instance should specify at least one of Event
+// and/or Payload.
+type LogData struct {
```

Comment:
I think this is an excellent idea... you previously mentioned opinionated recipes for doing various common tracing-related tasks, which is a related excellent idea.